sdk/test-utils/recorder-new/test/testProxyTests.spec.ts

```diff
@@ -94,6 +94,24 @@ function getTestServerUrl() {
       );
     });
 
+    it("sample_response with random string in path", async () => {
+      await recorder.start({ envSetupForPlayback: {} });
+
+      if (!isPlaybackMode()) {
+        recorder.variables["random-1"] = `random-${Math.ceil(Math.random() * 1000 + 1000)}`;
+        recorder.variables["random-2"] = "known-string";
+      }
+
+      await makeRequestAndVerifyResponse(
```

Comment:
Could you explain what this is verifying? I see a variable is set, the variable is used in the request path, and test server seems always return "I am the answer!" but am not clear what's involved

Reply:
oh, in playback mode, these variables will be available?

Reply:
`makeRequestAndVerifyResponse` is a generic method and is applied for all the tests in this test suite.
It was initially created for the rest of the tests in the describe block.
All it does is make the request and check the response with what we are supposed to get.

All tests are run in the record, playback, and live modes in sequence in the same run.

The idea of the test is that the random string won't change in playback mode, and it'll reuse whatever has been created in the record mode.

Any suggestions on how I can make this clear? or improve the test?

Reply:
This is fine. setting is in record mode but reading is in playback mode so I don't think there is a straight-forward way to test them as a whole. but end-to-end is working so we are good.